sympy/ntheory/generate.py

```diff
@@ -219,10 +242,31 @@ def prime(nth):
         primerange : Generate all primes in a given range
         primepi : Return the number of primes less than or equal to n
     """
-    n = as_int(nth)
+    n = int(nth)
     if n < 1:
         raise ValueError("nth must be a positive integer; prime(1) == 2")
-    return sieve[n]
+    if n<=2:
+        return n+1
+    from sympy.functions.special.error_functions import li
+    from sympy.functions.elementary.exponential import log
+    a = 2 # Lower bound for binary search
+    b = int(n*(log(n).evalf() + log(log(n).evalf()).evalf())) + 100 # Upper bound for the search.
+
+    while a < b:
+        mid = (a + b) >> 1
+        x = li(mid).evalf()
```

Comment:
It doesn't affect the answer.
About speed,I checked by changing it in my pc, and both take the same time.